pandas/core/groupby/generic.py

```diff
@@ -254,8 +254,11 @@ def _aggregate_item_by_item(self, func, *args, **kwargs):
                 data = obj[item]
                 colg = SeriesGroupBy(data, selection=item,
                                      grouper=self.grouper)
-                result[item] = self._try_cast(
-                    colg.aggregate(func, *args, **kwargs), data)
+
```

Comment:
instead of this, I think you can add to:

``cython_cast_blacklist = frozenset(['rank', 'count', 'size'])``

in pandas/core/groupby/base.py

Reply:
I will need to investigate it more deeply. 
First attempt with your suggestion did not work as expected. Maybe it is because _aggregate_item_by_item is called in special case when other fails (https://github.com/pandas-dev/pandas/blob/master/pandas/core/groupby/groupby.py#L622)?

Reply:
Hmm the point of that variable is to prevent things from getting cast to match their original dtype, so agreed also seems like the logical place for this. What failures is that causing?

Reply:
I hope, I found the suggested solution - please check: https://github.com/pandas-dev/pandas/pull/25531/commits/f23e884db3d44b8a8e8fff99e8f6eea4f3b1a786

Reply:
Thanks for checking on this. So I think while this works it's somewhat confusing that we are using `_transform_should_cast` in an aggregate method - would it be possible to just make a generic `_should_cast` function that can be shared across aggregations / transformations to prevent this from happening?

Reply:
this should already be done in _try_cast

Reply:
> this should already be done in _try_cast

I am a newbie but similar approach is used on another place (use ```_transform_should_cast``` before ```_try_cast``` - e.g. https://github.com/pandas-dev/pandas/blob/169a56a33fb7f4fad447e70869599a99832eeee5/pandas/core/groupby/groupby.py#L829). So now, It seems that such check is not included in ```_try_cast```. Am I wrong?

I found that ```_try_cast``` has parameter ```numeric_only```. I tried to use it - it works correctly with dates, but not with floats (as expected). So this is not the way.